tests/expectations/metrics/test_core.py

```diff
@@ -1096,15 +1096,85 @@ def test_table_metric_pd(caplog):
 
 
 def test_column_pairs_equal_metric_pd():
-    df = pd.DataFrame({"a": [1, 2, 3, 3], "b": [1, 2, 3, 3]})
+    df = pd.DataFrame({"a": [1, 2, 3, 3, 5, None], "b": [1, 2, 3, 4, None, None]})
     engine = PandasExecutionEngine(batch_data_dict={"my_id": df})
-    desired_metric = MetricConfiguration(
+
+    metrics: dict = {}
+
+    table_columns_metric: MetricConfiguration
+    results: dict
+
+    table_columns_metric, results = get_table_columns_metric(engine=engine)
+    metrics.update(results)
+
+    condition_metric = MetricConfiguration(
         metric_name="column_pair_values.equal.condition",
-        metric_domain_kwargs={"column_A": "a", "column_B": "b"},
-        metric_value_kwargs=dict(),
+        metric_domain_kwargs={
+            "column_A": "a",
+            "column_B": "b",
+            "ignore_row_if": "both_values_are_missing",
+        },
+        metric_dependencies={
+            "table.columns": table_columns_metric,
+        },
+    )
+    results = engine.resolve_metrics(
+        metrics_to_resolve=(condition_metric,),
+        metrics=metrics,
+    )
+    metrics.update(results)
+
+    assert (
+        results[condition_metric.id][0]
+        .reset_index(drop=True)
+        .equals(pd.Series([False, False, False, True, True]))
+    )
```

Comment:
Do we want to throw any assertion messages here?

Reply:
It wouldn't follow the current pattern used elsewhere. I think the developer should be able to follow the logic whereby each test changes based on the `ignore_row_if` argument. If more than one test is failing it is due to some other reason that would require a very generic message.